Ladybird/CMakeLists.txt

```diff
@@ -121,6 +121,11 @@ target_include_directories(headless-browser PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
 target_include_directories(headless-browser PRIVATE ${LADYBIRD_SOURCE_DIR}/Userland/)
 target_link_libraries(headless-browser PRIVATE AK LibCore LibWeb LibWebView LibWebSocket LibCrypto LibFileSystem LibHTTP LibImageDecoderClient LibJS LibGfx LibMain LibTLS LibIPC LibDiff LibProtocol LibURL)
 
+execute_process(COMMAND git rev-parse --short HEAD
+                OUTPUT_VARIABLE LADYBIRD_COMMIT_HASH
+                OUTPUT_STRIP_TRAILING_WHITESPACE)
+target_compile_definitions(LibWeb PUBLIC LADYBIRD_COMMIT_HASH="${LADYBIRD_COMMIT_HASH}")
```

Comment:
I guess one reason to not do this would be that it makes reproducible builds a bit more interesting to get right.

I know nico has commented in the past about using some kind of content hash as a version identifier instead? But that would need an actually reproducible, hermetic build :thinking: . 

For serenity, we generate the kernel version something like this, but in the GN build I created a nicer (imo) python script to create a Version.h header instead.

https://github.com/SerenityOS/serenity/blob/26555baf53fd89622c564d3e87582088a3fb09e8/Meta/gn/secondary/Kernel/generate_version_header.py

One issue with using a public target compile defintion on LibWeb is that now whenever the commit hash changes, we will be forced to rebuild every target that depends on libweb (because its compile flags changed).

Reply:
The CMake version runs a script that generates a header file, which is then added as a dependency to the kernel target.

https://github.com/SerenityOS/serenity/blob/26555baf53fd89622c564d3e87582088a3fb09e8/Kernel/CMakeLists.txt#L603-L613

https://github.com/SerenityOS/serenity/blob/26555baf53fd89622c564d3e87582088a3fb09e8/Kernel/generate-version-file.sh

Reply:
Nico's comment that I was thinking of is here: https://github.com/SerenityOS/serenity/pull/15068#issuecomment-1230494660

> Embedding git revisions in binaries means that two identical source trees except for a comment change somewhere produce different binaries.
>
> The cool way to handle this is to have deterministic builds and print the hash of the build outputs, not the inputs 😛

Reply:
Yeah, I'm not particularly fond of adding commit hash here either. A lot of the time, I'll be on a branch anyways with commits that aren't on `master`. And if we need to know what commit someone is on, it's way faster to just get it from e.g. `git log` than to build & run the browser and navigate to about:version.

When we have released versions with version numbers, it'll be easy to add that to the about:version page.